homeseer/hs-device.js

```diff
@@ -16,28 +16,50 @@ module.exports = function(RED) {
 			node.status({fill: "red", shape: "ring", text: "error while getting status"});
 		});
         
-        node.on('input', function(msg) {
+    node.on('input', function(msg) {
 			//console.log(node);
 			if(typeof msg.payload != 'undefined') {
-				if(typeof msg.payload.value != 'undefined') {
-					node.server.controlDeviceByValue(node.ref, msg.payload.value).then( data => {
-						msg.payload = data;
-						node.send(msg);
-					}).catch(err => {
-						msg.payload = err;
-						node.send(msg);
-					});
-				} else if(typeof msg.payload.status != 'undefined') {
-					node.server.controlDeviceByLabel(node.ref, msg.payload.status).then( data => {
-						msg.payload = data;
-						node.send(msg);
-					}).catch(err => {
-						msg.payload = err;
-						node.send(msg);
-					});
-				}
+        console.log(typeof msg.topic);
+        if(typeof msg.topic == 'string') msg.topic = msg.topic.toLowerCase();
+        if(!msg.topic || msg.topic == 'control') {
+          if(typeof msg.payload.value != 'undefined') {
+  					node.server.controlDeviceByValue(node.ref, msg.payload.value).then( data => {
+  						msg.payload = data;
+  						node.send(msg);
+  					}).catch(err => {
+  						msg.payload = err;
+  						node.send(msg);
+  					});
+  				} else if(typeof msg.payload.status != 'undefined') {
+  					node.server.controlDeviceByLabel(node.ref, msg.payload.status).then( data => {
+  						msg.payload = data;
+  						node.send(msg);
+  					}).catch(err => {
+  						msg.payload = err;
+  						node.send(msg);
+  					});
+          }
+        } else if(msg.topic == 'update') {
+          if(typeof msg.payload.value != 'undefined') {
+  					node.server.setDeviceValue(node.ref, msg.payload.value).then( data => {
+  						msg.payload = data;
+  						node.send(msg);
+  					}).catch(err => {
+  						msg.payload = err;
+  						node.send(msg);
+  					});
+          } else if(typeof msg.payload.status != 'undefined') {
```

Comment:
in case of an update, user may want to update both the value and status at the same time, so here I would replace the "else if" by a if

Reply:
> in case of an update, user may want to update both the value and status at the same time, so here I would replace the "else if" by a if

I had the same thought but wanted to run it past you first but it slipped my mind. Thanks for catching it. Easy change.